### ISSUE TITLE
Feature Request: inputstream

### ISSUE THREAD

**Issue by tedorf** [koying/SPMC#197]
Dear Chris,

thanks for all the work you put into XMBC/KODI and SPMC!

I saw on kodinerds.net, that some developers managed to use the new inputstream interface (inputstream.mpd) of kodi 17 to play amazon prime videos inside kodi (using widewine). It currently works on linux and windows, but nobody looked into an android option yet. Is this planned for a future version of spmc?

https://github.com/liberty-developer/inputstream.mpd
https://github.com/liberty-developer/plugin.video.prime_instant

Best,
Niko

**Comment by nevrion** [comment]
Oh yeah..that would be great

**Comment by strikerboy1024** [comment]
I agree, could be essential for the future
https://github.com/mapfau/inputstream.smoothstream
https://github.com/mapfau/inputstream.mpd

**Comment by koying** [comment]
What are the use cases, currently?

**Comment by tedorf** [comment]
Some people in the German kodinerds.net forum developed two video addons, one for amazon prime and the other one for skygo, that allow native playback of drm protected contents. These addons use the inputstream.mpd (amazon) and inputstream.smoothstream (skygo) addons (see links above) that just made it to the kodi-master (nightlies of kodi 17) . In addition, they are currently working on a video addon for netflix utilizing the same inputstream addons in the master.

This is the forum post that explains everything:
http://www.kodinerds.net/index.php/Thread/49988-kodi-17-plugin-amazon-prime-reloaded/
or http://www.kodinerds.net/index.php/Thread/50836-beta-kodi-17-SkyGo-Plugin

Unfortunately, it's in German. I am not sure if you speak German. If not, maybe google translate can help to understand. It now works on every device but android, with the current kodi 17 nightly + inputstream activated (which is now part of the kodi-master) and these addons installed.

**Comment by dogfight76** [comment]
Amazon Prime Video on Nvidia Shield TV......that would be great !! Pls make/try/do it

**Comment by tedorf** [comment]
@koying Are there any news on this?

**Comment by strikerboy1024** [comment]
@koying What do you think for general? 

Now, there are 3 Addons:
- Amazon Prime
- SkyGo
- Maxdome

...more will come, but not on Android till now. That's very sad for the german market because Kodi/SPMC loses attraction.

I think Liberty-developer https://github.com/liberty-developer would give support. So if it's possible...

**Comment by kaileu** [comment]
@koying the uses cases are that videoplayback was outsourced to binary addons. These Binary Addons implements adaptive streaming and so on.

**Comment by Uukrull** [comment]
@koying, now that you added mpd support to SPMC, will widevine be supported in the future (Amazon Video, Netflix), or is that impossible on Android?

**Comment by koying** [comment]
Very possible. Widevine is built-in in Android, so no need of hacky DLL's...
Problem #1 is that there is no AmazonPrime in my country...

**Comment by Uukrull** [comment]
Fantastic news!
You can try the 14 days free trial of [getflix](https://www.getflix.com.au/) and the Prime free month from amazon.com. I have a getflix account and it worked like a charm with Krypton and the Amazon Video addon from kodinerds.

**Comment by kaileu** [comment]
Could you also implement hls?

**Comment by koying** [comment]
What about HLS? Isn't that plain M3U8?

**Comment by kaileu** [comment]
Yes it is but adaptive Streaming isn't supported or did i just missed  
this feature?

Mit freundlichen Grüßen
Kai Leutner

**Comment by tedorf** [comment]
It's great to hear this is possible! Independent from the fact that I am
one of the users who would actually use this function, I also would love
for it to be realized just because many users in forums talk about Kodi for
Android being dead in the future because inputstream will not be added
(while many add-ons are currently developed that use it). So, if it's in
the end actually easier for users because widevine is part of Android (and
users don't need to copy hacky dll's), I am looking forward to the
reactions in the forums ;-)

**Comment by wrxtasy** [comment]
As a follow up to this, Android AMLogic S905 devices with the correct DRM licences such as the Mi Box and the S905 WeTek Hub / Play2 can also utilise the Kodi plugins listed at the top of the page for 1080p Amazon Video using a special AML Kodi 18 Leia test version.
